lib/Server.js

```diff
@@ -3255,9 +3255,70 @@ class Server {
    * @param {string | string[]} watchPath
    * @param {WatchOptions} [watchOptions]
    */
-  watchFiles(watchPath, watchOptions) {
+  watchFiles(watchPath, watchOptions = {}) {
     const chokidar = require("chokidar");
-    const watcher = chokidar.watch(watchPath, watchOptions);
+
+    const watchPathArr = Array.isArray(watchPath) ? watchPath : [watchPath];
+
+    if (watchOptions.disableGlobbing !== true) {
+      const isGlob = require("is-glob");
+      const watchPathGlobs = watchPathArr.filter((p) => isGlob(p));
+
+      // No need to do all this work when no globs are used
+      if (watchPathGlobs.length > 0) {
+        const globParent = require("glob-parent");
+        const picomatch = require("picomatch");
+
+        watchPathGlobs.forEach((p) => {
+          watchPathArr[watchPathArr.indexOf(p)] = globParent(p);
+        });
+
+        const matcher = picomatch(watchPathGlobs, {
+          cwd: watchOptions.cwd,
+          dot: true,
```

Comment:
`watchOptions.dot` (by default `false`)

It might even be better to pass all `watchOptions`

Reply:
reading the code from chokidar v3 it seems this defaulted to `true`. It was also not an option you could pass to `watchOptions`.

See here a constant is passed to anymatch:

https://github.com/paulmillr/chokidar/blob/7c50e25d10a497ce4409f6e52eb630f0d7647b97/index.js#L214

And the constant is defined as `{ dot: true }`

https://github.com/paulmillr/chokidar/blob/7c50e25d10a497ce4409f6e52eb630f0d7647b97/lib/constants.js#L56

The readme doesnt mention `dot` as an option: 

https://github.com/paulmillr/chokidar/tree/3.6.0?tab=readme-ov-file#api

So I dont think we should pass any other options besides that one here.